src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpDoNotGuardDictionaryRemoveByContainsKeyFixer.cs

```diff
@@ -16,11 +17,11 @@ public sealed class CSharpDoNotGuardDictionaryRemoveByContainsKeyFixer : DoNotGu
     {
         protected override bool SyntaxSupportedByFixer(SyntaxNode conditionalSyntax)
         {
-            if (conditionalSyntax is ConditionalExpressionSyntax conditionalExpressionSyntax)
```

Comment:
I looked at the code for the analyzer and it seems the first additional location for the diagnostic is the conditional operation syntax, which can either be an if statement syntax OR a conditional expression syntax (ternary operator): https://github.com/dotnet/roslyn-analyzers/blob/adcbeee529598d963cb9659bf3ea3eab90ca1455/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKey.cs#L81

I think your change here to exclude `ConditionalExpressionSyntax` has identified a test hole. Please add a test for ternary operators to confirm the old behavior, the new behavior and which one is more preferable.

Reply:
The analyzer never triggered for conditional expression syntax (even though the C# fixer had that case in mind).
This is due to the following switch always matching the default case for ternary operators:

https://github.com/dotnet/roslyn-analyzers/blob/20ba369077a407bf826651711b7f6d757ecbfcd2/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/DoNotGuardDictionaryRemoveByContainsKey.cs#L83-L114

I am not sure how a conditional expression syntax would look like that we want to fix.
Something like this comes to my mind:

```C#
bool removed = MyDictionary.ContainsKey("Key") ? MyDictionary.Remove("Key") : false;
```

But the above already uses the return value of `Dictionary.Remove`, so I think it is not likely someone would write it as a ternary expression.

---

I will keep the occurrence of `ConditionalExpressionSyntax` removed in the C# fixer (as it was not supported anyway) and add a test as you suggested.